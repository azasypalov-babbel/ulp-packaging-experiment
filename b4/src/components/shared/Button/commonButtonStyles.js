import { css } from 'styled-components';
import { setSaturation, setLightness, transparentize } from 'polished';
import { compose } from '../../../lib/compose';

const buttonPaddingX = '1rem';
export const buttonBorderWidth = '1px';

export const sizeStyles = css`
  height: 3rem;
  min-width: ${({ small, auto }) => {
    if (small) return '4.6rem';
    if (auto) return 'auto';
    return '7.5rem';
  }};
`;

export const paddingWithBorderStyles = (
  borderWidth = buttonBorderWidth,
  paddingX = buttonPaddingX
) => css`
  padding: 0 ${paddingX};

  &:hover, &:focus {
    padding: 0 calc(${borderWidth} + ${paddingX});
  }

  &:disabled:hover {
    padding: 0 ${paddingX};
  }

  &:active, &.is-active {
    padding: 0 calc(${borderWidth} + ${paddingX});
  }
`;

const transformShadowColor = (transparency) => compose(
  setSaturation(1),
  setLightness(0.2),
  transparentize(transparency)
);

export const commonShadowStyles = (shadowColor) => css`
  &:hover {
    box-shadow:
      inset 0 -2px 0 0 ${transformShadowColor(0.7)(shadowColor)};
  }

  &:active, &.is-active {
    box-shadow:
      inset 0 2px 0 0 ${transformShadowColor(0.5)(shadowColor)},
      inset 0 0 2px 0 ${transformShadowColor(0.8)(shadowColor)};
  }
`;

const commonButtonStyles = css`
  display: inline-block;
  box-sizing: border-box;
  
  ${sizeStyles}

  justify-content: center;
  align-items: center;

  font-family: ${(props) => props.theme.cascada.fontMilliardSemi};
  font-size: 1rem;
  line-height: 1;
  letter-spacing: 0.025rem;

  border-radius: 0.25rem;

  transition:
    box-shadow 100ms,
    background-color 100ms;

  white-space: nowrap;
  cursor: pointer;
  user-select: none;

  /* disabling double-tap to zoom */
  touch-action: manipulation;

  ${paddingWithBorderStyles()};

  &:focus {
    outline: none;
  }
`;

export default commonButtonStyles;
