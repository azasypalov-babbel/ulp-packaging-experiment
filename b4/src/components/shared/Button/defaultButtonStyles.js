import { css } from 'styled-components';
import { buttonBorderWidth, commonShadowStyles } from './commonButtonStyles';

const defaultButtonStyles = css`
  background-color: ${(props) => props.theme.cascada.white};
  color: ${(props) => props.theme.cascada.storm};

  border: ${buttonBorderWidth} solid ${(props) => props.theme.cascada.pastelStorm};

  &:hover, &:focus {
    border: none;
    background-color: ${(props) => props.theme.cascada.pastelElectric};
  }

  &:active, &.is-active {
    border: none;
    background-color: ${(props) => props.theme.cascada.survivalBlue};
    color: ${(props) => props.theme.cascada.white};
  }

  ${(props) => commonShadowStyles(props.theme.cascada.storm)}
`;

export default defaultButtonStyles;

