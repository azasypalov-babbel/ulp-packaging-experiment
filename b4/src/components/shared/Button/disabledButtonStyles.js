import { css } from 'styled-components';
import { buttonBorderWidth } from './commonButtonStyles';

const disabledButtonStyles = css`
  background-color: ${(props) => props.theme.cascada.platinium};
  color: ${(props) => props.theme.cascada.platiniumStorm};

  border: ${buttonBorderWidth} solid ${(props) => props.theme.cascada.pastelStorm};

`;

export default disabledButtonStyles;
