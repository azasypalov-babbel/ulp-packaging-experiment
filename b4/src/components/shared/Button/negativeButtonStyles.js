import { css } from 'styled-components';
import { buttonBorderWidth, commonShadowStyles } from './commonButtonStyles';

const negativeButtonStyles = css`
  background-color: ${(props) => props.theme.color.semantic.feedback.negative.background};
  color: ${(props) => props.theme.color.semantic.feedback.negative.foreground};

  border: ${buttonBorderWidth} solid ${(props) => props.theme.color.semantic.feedback.negative.border};

  & svg {
    color: ${(props) => props.theme.color.semantic.feedback.negative.foreground};
  }

  &:hover, &:focus {
    border: none;
  }

  &:active, &.is-active {
    border: none;
    background-color: ${(props) => props.theme.color.semantic.feedback.negative.border};
    color: ${(props) => props.theme.cascada.white};

    & svg {
      color: ${(props) => props.theme.cascada.white};
    }
  }

  ${(props) => commonShadowStyles(props.theme.color.semantic.feedback.negative.background)}
`;

export default negativeButtonStyles;

