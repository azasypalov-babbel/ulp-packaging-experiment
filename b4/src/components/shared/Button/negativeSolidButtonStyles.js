import { css } from 'styled-components';
import { commonShadowStyles } from './commonButtonStyles';
import { lighten } from 'polished';

const primaryButtonStyles = css`
  background-color: ${(props) => props.theme.cascada.red};
  color: ${(props) => props.theme.cascada.white};

  border: none;

  &:hover, &:focus {
    background-color: ${(props) => lighten(0.05, props.theme.cascada.red)};
  }

  &:active, &.is-active {
    background-color: ${(props) => props.theme.cascada.red};
  }

  ${(props) => commonShadowStyles(props.theme.cascada.red)};
`;

export default primaryButtonStyles;
