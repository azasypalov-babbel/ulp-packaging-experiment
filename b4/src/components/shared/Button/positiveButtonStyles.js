import { css } from 'styled-components';
import { buttonBorderWidth, commonShadowStyles } from './commonButtonStyles';

const positiveButtonStyles = css`
  background-color: ${(props) => props.theme.color.semantic.feedback.positive.background};
  color: ${(props) => props.theme.color.semantic.feedback.positive.foreground};

  border: ${buttonBorderWidth} solid ${(props) => props.theme.color.semantic.feedback.positive.border};

  & svg {
    color: ${(props) => props.theme.color.semantic.feedback.positive.foreground};
  }

  &:hover, &:focus {
    border: none;
  }

  &:active, &.is-active {
    border: none;
    background-color: ${(props) => props.theme.color.semantic.feedback.positive.border};
    color: ${(props) => props.theme.cascada.white};

    & svg {
      color: ${(props) => props.theme.cascada.white};
    }
  }

  ${(props) => commonShadowStyles(props.theme.color.semantic.feedback.positive.background)}
`;

export default positiveButtonStyles;

