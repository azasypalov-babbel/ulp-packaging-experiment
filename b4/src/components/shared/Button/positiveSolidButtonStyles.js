import { css } from 'styled-components';
import { commonShadowStyles } from './commonButtonStyles';
import { lighten } from 'polished';

const primaryButtonStyles = css`
  background-color: ${(props) => props.theme.cascada.teal};
  color: ${(props) => props.theme.cascada.white};

  border: none;

  &:hover, &:focus {
    background-color: ${(props) => lighten(0.1, props.theme.cascada.teal)};
  }

  &:active, &.is-active {
    background-color: ${(props) => props.theme.cascada.teal};
  }

  ${(props) => commonShadowStyles(props.theme.cascada.teal)};
`;

export default primaryButtonStyles;
