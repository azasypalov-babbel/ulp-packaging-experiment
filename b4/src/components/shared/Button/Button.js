import styled from 'styled-components';
import PropTypes from 'prop-types';

import commonButtonStyles from './commonButtonStyles';
import defaultButtonStyles from './defaultButtonStyles';
import disabledButtonStyles from './disabledButtonStyles';
import primaryButtonStyles from './primaryButtonStyles';
import positiveButtonStyles from './positiveButtonStyles';
import negativeButtonStyles from './negativeButtonStyles';
import positiveSolidButtonStyles from './positiveSolidButtonStyles';
import negativeSolidButtonStyles from './negativeSolidButtonStyles';
import withKeypress from '../withKeypress';

const StyledButton = styled.button.attrs(
  ({ pressed }) => ({ className: pressed ? 'is-active' : '' })
)`
  ${commonButtonStyles};
  ${({ disabled, positive, negative, primary, positiveSolid, negativeSolid }) => {
    if (disabled) return disabledButtonStyles;
    if (primary) return primaryButtonStyles;
    if (negative) return negativeButtonStyles;
    if (positive) return positiveButtonStyles;
    if (positiveSolid) return positiveSolidButtonStyles;
    if (negativeSolid) return negativeSolidButtonStyles;
    return defaultButtonStyles;
  }};
`;


export const Button = StyledButton;

Button.propTypes = {
  primary: PropTypes.bool,
  disabled: PropTypes.bool,
  pressed: PropTypes.bool
};

Button.defaultProps = {
  type: 'button'
};

export default withKeypress(Button);
