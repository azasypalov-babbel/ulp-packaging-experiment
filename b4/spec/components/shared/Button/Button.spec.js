import React from 'react';
import { Button } from '../../../../src/components/shared/Button/Button';
import { mountWithTheme } from '../themeMock';

describe('<Button />', () => {
  const clickSpy = jest.fn();
  beforeEach(() => {
    clickSpy.mockClear();
  });

  test('it renders', () => {
    const wrapper = mountWithTheme(<Button />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('when button is pressed', () => {
    test('it renders with is-active class', () => {
      const wrapper = mountWithTheme(<Button pressed />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('<Button disabled />', () => {
    test('it renders', () => {
      const wrapper = mountWithTheme(<Button disabled />);
      expect(wrapper).toMatchSnapshot();
    });

    test('it does not trigger onClick', () => {
      const wrapper = mountWithTheme(<Button disabled onClick={clickSpy} />);
      wrapper.simulate('click');

      expect(clickSpy).not.toHaveBeenCalled();
    });
  });

  describe('<Button primary />', () => {
    test('it renders', () => {
      const wrapper = mountWithTheme(<Button primary />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('<Button positive />', () => {
    test('it renders', () => {
      const wrapper = mountWithTheme(<Button positive />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('<Button negative />', () => {
    test('it renders', () => {
      const wrapper = mountWithTheme(<Button negative />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('<Button positiveSolid />', () => {
    test('it renders', () => {
      const wrapper = mountWithTheme(<Button positiveSolid />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('<Button negativeSolid />', () => {
    test('it renders', () => {
      const wrapper = mountWithTheme(<Button negativeSolid />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
