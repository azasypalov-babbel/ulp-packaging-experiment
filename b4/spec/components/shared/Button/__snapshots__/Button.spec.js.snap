// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> <Button disabled /> it renders 1`] = `
.c0 {
  display: inline-block;
  box-sizing: border-box;
  height: 3rem;
  min-width: 7.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: Milliard-SemiBold,sans-serif;
  font-size: 1rem;
  line-height: 1;
  -webkit-letter-spacing: 0.025rem;
  -moz-letter-spacing: 0.025rem;
  -ms-letter-spacing: 0.025rem;
  letter-spacing: 0.025rem;
  border-radius: 0.25rem;
  -webkit-transition: box-shadow 100ms, background-color 100ms;
  transition: box-shadow 100ms, background-color 100ms;
  white-space: nowrap;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  touch-action: manipulation;
  padding: 0 1rem;
  background-color: #e1e5e7;
  color: #9aa7b3;
  border: 1px solid #becedd;
}

.c0:hover,
.c0:focus {
  padding: 0 calc(1px + 1rem);
}

.c0:disabled:hover {
  padding: 0 1rem;
}

.c0:active,
.c0.is-active {
  padding: 0 calc(1px + 1rem);
}

.c0:focus {
  outline: none;
}

<styled.button
  disabled={true}
  type="button"
>
  <button
    className="c0"
    disabled={true}
    type="button"
  />
</styled.button>
`;

exports[`<Button /> <Button negative /> it renders 1`] = `
.c0 {
  display: inline-block;
  box-sizing: border-box;
  height: 3rem;
  min-width: 7.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: Milliard-SemiBold,sans-serif;
  font-size: 1rem;
  line-height: 1;
  -webkit-letter-spacing: 0.025rem;
  -moz-letter-spacing: 0.025rem;
  -ms-letter-spacing: 0.025rem;
  letter-spacing: 0.025rem;
  border-radius: 0.25rem;
  -webkit-transition: box-shadow 100ms, background-color 100ms;
  transition: box-shadow 100ms, background-color 100ms;
  white-space: nowrap;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  touch-action: manipulation;
  padding: 0 1rem;
  background-color: #ffdcdc;
  color: #284b7c;
  border: 1px solid #ff4d4d;
}

.c0:hover,
.c0:focus {
  padding: 0 calc(1px + 1rem);
}

.c0:disabled:hover {
  padding: 0 1rem;
}

.c0:active,
.c0.is-active {
  padding: 0 calc(1px + 1rem);
}

.c0:focus {
  outline: none;
}

.c0 svg {
  color: #284b7c;
}

.c0:hover,
.c0:focus {
  border: none;
}

.c0:active,
.c0.is-active {
  border: none;
  background-color: #ff4d4d;
  color: #ffffff;
}

.c0:active svg,
.c0.is-active svg {
  color: #ffffff;
}

.c0:hover {
  box-shadow: inset 0 -2px 0 0 setSaturation(1,setLightness(0.2,transparentize(0.7,#ffdcdc)));
}

.c0:active,
.c0.is-active {
  box-shadow: inset 0 2px 0 0 setSaturation(1,setLightness(0.2,transparentize(0.5,#ffdcdc))), inset 0 0 2px 0 setSaturation(1,setLightness(0.2,transparentize(0.8,#ffdcdc)));
}

<styled.button
  negative={true}
  type="button"
>
  <button
    className="c0"
    type="button"
  />
</styled.button>
`;

exports[`<Button /> <Button negativeSolid /> it renders 1`] = `
.c0 {
  display: inline-block;
  box-sizing: border-box;
  height: 3rem;
  min-width: 7.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: Milliard-SemiBold,sans-serif;
  font-size: 1rem;
  line-height: 1;
  -webkit-letter-spacing: 0.025rem;
  -moz-letter-spacing: 0.025rem;
  -ms-letter-spacing: 0.025rem;
  letter-spacing: 0.025rem;
  border-radius: 0.25rem;
  -webkit-transition: box-shadow 100ms, background-color 100ms;
  transition: box-shadow 100ms, background-color 100ms;
  white-space: nowrap;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  touch-action: manipulation;
  padding: 0 1rem;
  background-color: #ff4d4d;
  color: #ffffff;
  border: none;
}

.c0:hover,
.c0:focus {
  padding: 0 calc(1px + 1rem);
}

.c0:disabled:hover {
  padding: 0 1rem;
}

.c0:active,
.c0.is-active {
  padding: 0 calc(1px + 1rem);
}

.c0:focus {
  outline: none;
}

.c0:hover,
.c0:focus {
  background-color: lighten(0.05,#ff4d4d);
}

.c0:active,
.c0.is-active {
  background-color: #ff4d4d;
}

.c0:hover {
  box-shadow: inset 0 -2px 0 0 setSaturation(1,setLightness(0.2,transparentize(0.7,#ff4d4d)));
}

.c0:active,
.c0.is-active {
  box-shadow: inset 0 2px 0 0 setSaturation(1,setLightness(0.2,transparentize(0.5,#ff4d4d))), inset 0 0 2px 0 setSaturation(1,setLightness(0.2,transparentize(0.8,#ff4d4d)));
}

<styled.button
  negativeSolid={true}
  type="button"
>
  <button
    className="c0"
    type="button"
  />
</styled.button>
`;

exports[`<Button /> <Button positive /> it renders 1`] = `
.c0 {
  display: inline-block;
  box-sizing: border-box;
  height: 3rem;
  min-width: 7.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: Milliard-SemiBold,sans-serif;
  font-size: 1rem;
  line-height: 1;
  -webkit-letter-spacing: 0.025rem;
  -moz-letter-spacing: 0.025rem;
  -ms-letter-spacing: 0.025rem;
  letter-spacing: 0.025rem;
  border-radius: 0.25rem;
  -webkit-transition: box-shadow 100ms, background-color 100ms;
  transition: box-shadow 100ms, background-color 100ms;
  white-space: nowrap;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  touch-action: manipulation;
  padding: 0 1rem;
  background-color: #e3f8f7;
  color: #284b7c;
  border: 1px solid #44cdc8;
}

.c0:hover,
.c0:focus {
  padding: 0 calc(1px + 1rem);
}

.c0:disabled:hover {
  padding: 0 1rem;
}

.c0:active,
.c0.is-active {
  padding: 0 calc(1px + 1rem);
}

.c0:focus {
  outline: none;
}

.c0 svg {
  color: #284b7c;
}

.c0:hover,
.c0:focus {
  border: none;
}

.c0:active,
.c0.is-active {
  border: none;
  background-color: #44cdc8;
  color: #ffffff;
}

.c0:active svg,
.c0.is-active svg {
  color: #ffffff;
}

.c0:hover {
  box-shadow: inset 0 -2px 0 0 setSaturation(1,setLightness(0.2,transparentize(0.7,#e3f8f7)));
}

.c0:active,
.c0.is-active {
  box-shadow: inset 0 2px 0 0 setSaturation(1,setLightness(0.2,transparentize(0.5,#e3f8f7))), inset 0 0 2px 0 setSaturation(1,setLightness(0.2,transparentize(0.8,#e3f8f7)));
}

<styled.button
  positive={true}
  type="button"
>
  <button
    className="c0"
    type="button"
  />
</styled.button>
`;

exports[`<Button /> <Button positiveSolid /> it renders 1`] = `
.c0 {
  display: inline-block;
  box-sizing: border-box;
  height: 3rem;
  min-width: 7.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: Milliard-SemiBold,sans-serif;
  font-size: 1rem;
  line-height: 1;
  -webkit-letter-spacing: 0.025rem;
  -moz-letter-spacing: 0.025rem;
  -ms-letter-spacing: 0.025rem;
  letter-spacing: 0.025rem;
  border-radius: 0.25rem;
  -webkit-transition: box-shadow 100ms, background-color 100ms;
  transition: box-shadow 100ms, background-color 100ms;
  white-space: nowrap;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  touch-action: manipulation;
  padding: 0 1rem;
  background-color: #44cdc8;
  color: #ffffff;
  border: none;
}

.c0:hover,
.c0:focus {
  padding: 0 calc(1px + 1rem);
}

.c0:disabled:hover {
  padding: 0 1rem;
}

.c0:active,
.c0.is-active {
  padding: 0 calc(1px + 1rem);
}

.c0:focus {
  outline: none;
}

.c0:hover,
.c0:focus {
  background-color: lighten(0.1,#44cdc8);
}

.c0:active,
.c0.is-active {
  background-color: #44cdc8;
}

.c0:hover {
  box-shadow: inset 0 -2px 0 0 setSaturation(1,setLightness(0.2,transparentize(0.7,#44cdc8)));
}

.c0:active,
.c0.is-active {
  box-shadow: inset 0 2px 0 0 setSaturation(1,setLightness(0.2,transparentize(0.5,#44cdc8))), inset 0 0 2px 0 setSaturation(1,setLightness(0.2,transparentize(0.8,#44cdc8)));
}

<styled.button
  positiveSolid={true}
  type="button"
>
  <button
    className="c0"
    type="button"
  />
</styled.button>
`;

exports[`<Button /> <Button primary /> it renders 1`] = `
.c0 {
  display: inline-block;
  box-sizing: border-box;
  height: 3rem;
  min-width: 7.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: Milliard-SemiBold,sans-serif;
  font-size: 1rem;
  line-height: 1;
  -webkit-letter-spacing: 0.025rem;
  -moz-letter-spacing: 0.025rem;
  -ms-letter-spacing: 0.025rem;
  letter-spacing: 0.025rem;
  border-radius: 0.25rem;
  -webkit-transition: box-shadow 100ms, background-color 100ms;
  transition: box-shadow 100ms, background-color 100ms;
  white-space: nowrap;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  touch-action: manipulation;
  padding: 0 1rem;
  background-color: #ff790f;
  color: #ffffff;
  border: none;
}

.c0:hover,
.c0:focus {
  padding: 0 calc(1px + 1rem);
}

.c0:disabled:hover {
  padding: 0 1rem;
}

.c0:active,
.c0.is-active {
  padding: 0 calc(1px + 1rem);
}

.c0:focus {
  outline: none;
}

.c0:hover,
.c0:focus {
  background-color: #ffaf00;
}

.c0:active,
.c0.is-active {
  background-color: #ff790f;
}

.c0:hover {
  box-shadow: inset 0 -2px 0 0 setSaturation(1,setLightness(0.2,transparentize(0.7,#9a4400)));
}

.c0:active,
.c0.is-active {
  box-shadow: inset 0 2px 0 0 setSaturation(1,setLightness(0.2,transparentize(0.5,#9a4400))), inset 0 0 2px 0 setSaturation(1,setLightness(0.2,transparentize(0.8,#9a4400)));
}

<styled.button
  primary={true}
  type="button"
>
  <button
    className="c0"
    type="button"
  />
</styled.button>
`;

exports[`<Button /> it renders 1`] = `
.c0 {
  display: inline-block;
  box-sizing: border-box;
  height: 3rem;
  min-width: 7.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: Milliard-SemiBold,sans-serif;
  font-size: 1rem;
  line-height: 1;
  -webkit-letter-spacing: 0.025rem;
  -moz-letter-spacing: 0.025rem;
  -ms-letter-spacing: 0.025rem;
  letter-spacing: 0.025rem;
  border-radius: 0.25rem;
  -webkit-transition: box-shadow 100ms, background-color 100ms;
  transition: box-shadow 100ms, background-color 100ms;
  white-space: nowrap;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  touch-action: manipulation;
  padding: 0 1rem;
  background-color: #ffffff;
  color: #284b7c;
  border: 1px solid #becedd;
}

.c0:hover,
.c0:focus {
  padding: 0 calc(1px + 1rem);
}

.c0:disabled:hover {
  padding: 0 1rem;
}

.c0:active,
.c0.is-active {
  padding: 0 calc(1px + 1rem);
}

.c0:focus {
  outline: none;
}

.c0:hover,
.c0:focus {
  border: none;
  background-color: #d1eefc;
}

.c0:active,
.c0.is-active {
  border: none;
  background-color: #738ce6;
  color: #ffffff;
}

.c0:hover {
  box-shadow: inset 0 -2px 0 0 setSaturation(1,setLightness(0.2,transparentize(0.7,#284b7c)));
}

.c0:active,
.c0.is-active {
  box-shadow: inset 0 2px 0 0 setSaturation(1,setLightness(0.2,transparentize(0.5,#284b7c))), inset 0 0 2px 0 setSaturation(1,setLightness(0.2,transparentize(0.8,#284b7c)));
}

<styled.button
  type="button"
>
  <button
    className="c0"
    type="button"
  />
</styled.button>
`;

exports[`<Button /> when button is pressed it renders with is-active class 1`] = `
.c0 {
  display: inline-block;
  box-sizing: border-box;
  height: 3rem;
  min-width: 7.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: Milliard-SemiBold,sans-serif;
  font-size: 1rem;
  line-height: 1;
  -webkit-letter-spacing: 0.025rem;
  -moz-letter-spacing: 0.025rem;
  -ms-letter-spacing: 0.025rem;
  letter-spacing: 0.025rem;
  border-radius: 0.25rem;
  -webkit-transition: box-shadow 100ms, background-color 100ms;
  transition: box-shadow 100ms, background-color 100ms;
  white-space: nowrap;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  touch-action: manipulation;
  padding: 0 1rem;
  background-color: #ffffff;
  color: #284b7c;
  border: 1px solid #becedd;
}

.c0:hover,
.c0:focus {
  padding: 0 calc(1px + 1rem);
}

.c0:disabled:hover {
  padding: 0 1rem;
}

.c0:active,
.c0.is-active {
  padding: 0 calc(1px + 1rem);
}

.c0:focus {
  outline: none;
}

.c0:hover,
.c0:focus {
  border: none;
  background-color: #d1eefc;
}

.c0:active,
.c0.is-active {
  border: none;
  background-color: #738ce6;
  color: #ffffff;
}

.c0:hover {
  box-shadow: inset 0 -2px 0 0 setSaturation(1,setLightness(0.2,transparentize(0.7,#284b7c)));
}

.c0:active,
.c0.is-active {
  box-shadow: inset 0 2px 0 0 setSaturation(1,setLightness(0.2,transparentize(0.5,#284b7c))), inset 0 0 2px 0 setSaturation(1,setLightness(0.2,transparentize(0.8,#284b7c)));
}

<styled.button
  pressed={true}
  type="button"
>
  <button
    className="c0 is-active"
    type="button"
  />
</styled.button>
`;
